gcp/api/integration_tests.py

```diff
@@ -582,6 +587,90 @@ def test_query_package_purl(self):
 
     self.assertEqual(set(), vulns_first.intersection(vulns_second))
 
+  @unittest.skipUnless(_LONG_TESTS, "Takes around 45 seconds running locally," +
+                       "enable when making a big change")
+  def test_all_possible_queries(self):
+    """Test all combinations of valid and invalid queries"""
+    semver_package = {'package': {'purl': 'pkg:cargo/crossbeam-utils'}}
+
+    semver_package_with_version = {
+        'package': {
+            'purl': 'pkg:cargo/crossbeam-utils@0.8.5'
+        }
+    }
+
+    nonsemver_package = {'package': {'purl': 'pkg:pypi/numpy'}}
+
+    nonsemver_package_with_version = {
+        'package': {
+            'purl': 'pkg:pypi/numpy@8.24.0'
+        }
+    }
+
+    pkg_ecosystem = [{'package': {'ecosystem': 'crates.io'}}, {}]
+
+    pkg_name = [{
+        'package': {
+            'name': 'crossbeam-utils'
+        }
+    }, {
+        'package': {
+            'name': 'numpy'
+        }
+    }, {}]
+
+    pkg_version = [{'package': {'version': '0.8.5'}}, {}]
+
+    commit = [{'commit': 'd374094d8c49b6b7d288f307e11217ec5a502391'}, {}]
+
+    purl_fields = [
+        semver_package, semver_package_with_version, nonsemver_package,
+        nonsemver_package_with_version, {}
+    ]
+
+    product = itertools.product(purl_fields, commit, pkg_version, pkg_name,
+                                pkg_ecosystem)
+
+    # itertools.product will produce duplicates, use set over the json to de-dup
+    combined_product = set()
+    for elem in product:
+      # Deep copy elem required since merge merges
+      # in-place to the first argument
+      elem = copy.deepcopy(elem)
+      functools.reduce(merge, elem)
+      combined_product.add(json.dumps(elem[0], sort_keys=True))
+
+    self.assertEqual(len(combined_product), 120)
+    with open('fixtures/api_query_response.txt') as h:
```

Comment:
Can we use https://github.com/google/osv.dev/blob/master/osv/tests.py#L54 ? 

That also makes it very easy to regenerate fixtures.

Reply:
Yep, now using the test module